test/e2e/token_inactivity_timeout_test.go

```diff
@@ -0,0 +1,336 @@
+}
+
+func createUser(t *testing.T, userClient *userclient.UserV1Client, userName, identity string) (types.UID, func()) {
+	user := &userapi.User{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userName,
+		},
+		Identities: []string{identity},
+	}
+
+	err := wait.PollImmediate(300*time.Millisecond, 2*time.Second, func() (bool, error) {
+		var err error
+		user, err = userClient.Users().Create(context.TODO(), user, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("failed to create user: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	require.NoError(t, err)
+	return user.UID, func() {
+		if err := userClient.Users().Delete(context.TODO(), userName, metav1.DeleteOptions{}); err != nil {
+			t.Logf("%v", err)
+		}
+	}
+}
+
+func createIdentity(t *testing.T, userClient *userclient.UserV1Client, userName, identityName, idpName string, uid types.UID) func() {
+	identity := &userapi.Identity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: identityName,
+		},
+		ProviderName:     idpName,
+		ProviderUserName: userName,
+		User: corev1.ObjectReference{
+			Name: userName,
+			UID:  uid,
+		},
+	}
+
+	err := wait.PollImmediate(300*time.Second, 2*time.Second, func() (bool, error) {
+		_, err := userClient.Identities().Create(context.TODO(), identity, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("failed to create user identity: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	require.NoError(t, err)
+	return func() {
+		if err := userClient.Identities().Delete(context.TODO(), identity.Name, metav1.DeleteOptions{}); err != nil {
+			t.Logf("%v", err)
+		}
+	}
+}
+
+func createOAuthClient(t *testing.T, oauthClientClient *oauthclient.OauthV1Client, oauthClientName string, redirectURIs []string, timeout *int32) func() {
+	oauthClient := &oauthapi.OAuthClient{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: oauthClientName,
+		},
+		Secret:                              "the-secret-for-oauth-client",
+		RedirectURIs:                        redirectURIs,
+		GrantMethod:                         "auto",
+		AccessTokenInactivityTimeoutSeconds: timeout,
+	}
+
+	err := wait.PollImmediate(300*time.Millisecond, 2*time.Second, func() (bool, error) {
+		_, err := oauthClientClient.OAuthClients().Create(context.TODO(), oauthClient, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("failed to create oauth client: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	require.NoError(t, err)
+	return func() {
+		if err := oauthClientClient.OAuthClients().Delete(context.TODO(), oauthClient.Name, metav1.DeleteOptions{}); err != nil {
+			t.Logf("%v", err)
+		}
+	}
+}
+
+func checkTokenAccess(t *testing.T,
+	kubeConfig *rest.Config,
+	userClient *userclient.UserV1Client,
+	oauthClientClient *oauthclient.OauthV1Client,
+	inactivityTimeout, oauthClientTimeout int32,
+	prefix string,
+	checkAccess func(*testing.T, *http.Transport, *http.Request, *oauthapi.OAuthAccessToken, *oauthapi.OAuthAccessToken, time.Duration)) {
```

Comment:
Usually we prefix them with `test` instead of `check`.

Reply:
How about having arbitrary many test funcs here? That would allow you to give them nice names further up.

Like:
```
checkTokenAccess(t, kubeConfig, userClient, oauthClientClient, 10, 0, "without-inactivity-timeout",
   tokenWithInactivityTimeoutExpired,
   tokenWithoutInactivityTimeoutWorks,
)
```